browser_use/browser/session.py

```diff
@@ -343,24 +350,21 @@ async def setup_browser_via_passed_objects(self) -> None:
 		"""Override to customize the set up of the connection to an existing browser"""
 
 		# 1. check for a passed Page object, if present, it always takes priority, set browser_context = page.context
-		self.browser_context = (self.agent_current_page and self.agent_current_page.context) or self.browser_context or None
+		if self.agent_current_page:
+			self.browser_context = self.agent_current_page.context
 
 		# 2. Check if the current browser connection is valid, if not clear the invalid objects
-		if self.browser_context:
-			try:
-				# Try to access a property that would fail if the context is closed
-				_ = self.browser_context.pages
-				# Additional check: verify the browser is still connected
-				if self.browser_context.browser and not self.browser_context.browser.is_connected():
-					self.browser_context = None
-			except Exception:
-				# Context is closed, clear it
-				self.browser_context = None
-
-		# 3. if we have a browser object but it's disconnected, clear it and the context because we cant use either
-		if self.browser and not self.browser.is_connected():
-			if self.browser_context and (self.browser_context.browser is self.browser):
-				self.browser_context = None
```

Comment:
this is still needed, there are cases where we get a zombie browser_context that looks connected that only `browser.is_connected()` can rule out